roles/validate_inventory/tasks/cluster.yml

```diff
@@ -5,11 +5,22 @@
       - openshift_full_version in supported_ocp_versions
     fail_msg: "We do not support that version of openshift"
 
-- name: Assert valid master configuration
+- name: Assert valid master configuration (HA)
   assert:
     that:
       - groups['masters'] | length >= 3
-    fail_msg: "There must be at least three masters defined."
+    fail_msg: "There must be at least three masters defined. To deploy SNO, define one master and no workers."
+  when: not is_valid_single_node_openshift_config
+
+- name: Assert API and Ingress VIPs are set correctly (SNO)
+  assert:
+    that:
+      - api_vip == hostvars[sno_hostname]['ansible_host']
+      - ingress_vip == hostvars[sno_hostname]['ansible_host']
+    fail_msg: "For SNO deployments, API and Ingress VIPs need to match the IP address (ansible_host) of the master node."
+  vars:
+    sno_hostname: "{{ groups['masters'][0] | default('super1') }}"
```

Comment:
At what point is the `when` evaluated? if it is before `vars` then I don't think you need a default here.

Reply:
You're right. Now that `is_valid_single_node_openshift_config` uses `groups['masters']` internally, the default here is redundant. I'll double-check that though.

Reply:
I've tested the removal of the default (it works) but forgot to commit the change. Fixed!